core/src/main/java/com/linecorp/armeria/server/ServerConfig.java

```diff
@@ -127,6 +136,9 @@
         this.meterRegistry = requireNonNull(meterRegistry, "meterRegistry");
         this.serviceLoggerPrefix = ServiceConfig.validateLoggerName(serviceLoggerPrefix, "serviceLoggerPrefix");
         this.accessLogWriter = requireNonNull(accessLogWriter, "accessLogWriter");
+        this.channelOptions = requireNonNull(ImmutableMap.copyOf(channelOptions), "channelOptions");
+        this.childChannelOptions = requireNonNull(ImmutableMap.copyOf(childChannelOptions),
+                                                  "childChannelOptions");
```

Comment:
This should be:

    ImmutableMap.copyOf(requireNonNull(channelOptions, "channelOptions"))